### ISSUE TITLE
Strange behaviour with CART on dataset of two columns

### ISSUE THREAD

**Issue by stefvanbuuren** [stefvanbuuren/mice#27]
Hi Stef, hi Karin,

I'm a Ph.D. student at the University of Michigan and I'm trying to run a simulation on imputing missing data in a minimal setting (2 variables with missing values).

When I'm using `meth = "cart"` I'm receiving the following error message:
```
"Error in model.frame.default(formula = yobs ~ ., data = cbind(yobs, xobs),  : 
  'data' must be a data.frame, not a matrix or an array "
```

I've already searched the Internet for several days and tried different parameter settings, but I still can't find a solution or explanation for this problem. I've included a minimal working example showing that mice with `meth="cart"` on a dataset with two variables does not work. This data situation is not a problem for `"rpart"` or `meth="norm.predict"` in mice. (Adding a third variable produces no error)

It would be great if you could help me with that or point me to resources where I can find an solution to that problem (or at least an answer why it is not working).

If you need any further information, please let me know.

Thanks,
Micha

## minimal example:

```
require("mice")
require("rpart")
a = rnorm(n = 200)
b = rnorm(n = 200)
c = rnorm(n = 200)

a[1:20] = NA
b[15:40] = NA
c[30:50] = NA

df1 = data.frame(a,b)
df2 = data.frame(a,b,c)

imp1 = mice(data = df1,m=1, meth = "cart", maxsurrogate=0, maxit = 1) # produces the described error
imp1 = mice(data = df2,m=1, meth = "cart", maxsurrogate=0, maxit = 1) # works fine

imp2 = mice(data = df1,m=1, meth = "norm.predict", maxsurrogate=0, maxit=1)
#complete(imp2)

tree1 = rpart(a~b, data=df1, maxsurrogate=0)
tree2 = rpart(b~a, data=df1, maxsurrogate=0)
```

**Comment by stefvanbuuren** [comment]
Thanks a lot. 
`R` never stops to surprise me. The quick solution is to add a column, but of course, this needs to be fixed. We'll have a look at it.

**Comment by stefvanbuuren** [comment]
Fixed in 2.32
